crates/bevy_ecs/src/world/deferred_world.rs

```diff
@@ -175,142 +175,137 @@ impl<'w> DeferredWorld<'w> {
-    ///
-    /// ## Single [`Entity`]
+    /// Retrieves an [`EntityMut`] that exposes read and write operations for the given `entity`.
+    /// This will panic if the `entity` does not exist. Use [`DeferredWorld::get_entity_mut`] if you want
+    /// to check for entity existence instead of implicitly panic-ing.
     ///
     /// ```
-    /// # use bevy_ecs::{prelude::*, world::DeferredWorld};
+    /// use bevy_ecs::{component::Component, world::World};
+    ///
     /// #[derive(Component)]
     /// struct Position {
     ///   x: f32,
     ///   y: f32,
     /// }
     ///
-    /// # let mut world = World::new();
-    /// # let entity = world.spawn(Position { x: 0.0, y: 0.0 }).id();
-    /// let mut world: DeferredWorld = // ...
-    /// #   DeferredWorld::from(&mut world);
-    ///
+    /// let mut world = World::new();
+    /// let entity = world.spawn(Position { x: 0.0, y: 0.0 }).id();
     /// let mut entity_mut = world.entity_mut(entity);
     /// let mut position = entity_mut.get_mut::<Position>().unwrap();
-    /// position.y = 1.0;
-    /// assert_eq!(position.x, 0.0);
+    /// position.x = 1.0;
     /// ```
+    #[inline]
+    pub fn entity_mut(&mut self, entity: Entity) -> EntityMut<'_> {
+        self.get_entity_mut(entity).unwrap()
+    }
+
+    /// Gets mutable access to multiple entities.
     ///
-    /// ## Array of [`Entity`]s
+    /// # Errors
+    ///
+    /// - [`EntityMutableFetchError::EntityDoesNotExist`] if any entities do not exist in the world
+    /// - [`EntityMutableFetchError::AliasedMutability`] if the same entity is specified multiple times
+    ///
+    /// # Examples
     ///
     /// ```
     /// # use bevy_ecs::{prelude::*, world::DeferredWorld};
-    /// #[derive(Component)]
-    /// struct Position {
-    ///   x: f32,
-    ///   y: f32,
-    /// }
-    ///
     /// # let mut world = World::new();
-    /// # let e1 = world.spawn(Position { x: 0.0, y: 0.0 }).id();
-    /// # let e2 = world.spawn(Position { x: 1.0, y: 1.0 }).id();
+    /// # let id1 = world.spawn_empty().id();
+    /// # let id2 = world.spawn_empty().id();
     /// let mut world: DeferredWorld = // ...
     /// #   DeferredWorld::from(&mut world);
     ///
-    /// let [mut e1_ref, mut e2_ref] = world.entity_mut([e1, e2]);
-    /// let mut e1_position = e1_ref.get_mut::<Position>().unwrap();
-    /// e1_position.x = 1.0;
-    /// assert_eq!(e1_position.x, 1.0);
-    /// let mut e2_position = e2_ref.get_mut::<Position>().unwrap();
-    /// e2_position.x = 2.0;
-    /// assert_eq!(e2_position.x, 2.0);
+    /// // Disjoint mutable access.
+    /// let [entity1, entity2] = world.get_many_entities_mut([id1, id2]).unwrap();
+    ///
+    /// // Trying to access the same entity multiple times will fail.
+    /// assert!(world.get_many_entities_mut([id1, id1]).is_err());
     /// ```
+    pub fn get_many_entities_mut<const N: usize>(
+        &mut self,
+        entities: [Entity; N],
+    ) -> Result<[EntityMut<'_>; N], EntityMutableFetchError> {
```

Comment:
I wonder if at some point we should provide a variant that returns `[Result<EntityMut<'_>, EntityMutableFetchError>; N]`. 🤔 Unfortunately `std` doesn't provide a `transpose` function for results <-> arrays, so we'd have to provide that function ourselves or make the user perform it themselves.